app/Helpers/util/string.php

```diff
@@ -29,11 +28,20 @@ function attributeEscape(?string $input): string
 
 function isValidUsername(?string $username): bool
 {
+    /**
+     * Allows:
+     *   - \p{L}: Any kind of letter from any language.
+     *   - \p{N}: Any kind of numeric character from any script.
+     *   - \s   : Spaces.
+     *   - \-_.  : Hyphens, underscores, and periods.
+     */
+    $allowedPattern = '/^[\p{L}\p{N}\s\-_.]*$/u';
```

Comment:
I think we want to continue to restrict user names to ASCII only. If they want Unicode characters, they can put those in the display name.